src/olympia/api/tests/test_authentication.py

```diff
@@ -0,0 +1,231 @@
+from calendar import timegm
+import mock
+import json
+
+from django.core.urlresolvers import reverse
+from django.test import RequestFactory
+
+import jwt
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from olympia.amo.helpers import absolutify
+from olympia.amo.tests import TestCase, WithDynamicEndpoints
+from olympia.api.authentication import (
+    JSONWebTokenAuthentication, JWTKeyAuthentication)
+from olympia.api.tests.test_jwt_auth import JWTAuthKeyTester
+from olympia.users.models import UserProfile
+
+
+class JWTKeyAuthTestView(APIView):
+    """
+    This is an example of a view that would be protected by
+    JWTKeyAuthentication, used in TestJWTKeyAuthProtectedView below.
+    """
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JWTKeyAuthentication]
+
+    def get(self, request):
+        return Response('some get response')
+
+    def post(self, request):
+        return Response({'user_pk': request.user.pk})
+
+
+class TestJWTKeyAuthentication(JWTAuthKeyTester):
+    fixtures = ['base/addon_3615']
+
+    def setUp(self):
+        super(TestJWTKeyAuthentication, self).setUp()
+        self.factory = RequestFactory()
+        self.auth = JWTKeyAuthentication()
+        self.user = UserProfile.objects.get(email='del@icio.us')
+
+    def request(self, token):
+        return self.factory.get('/', HTTP_AUTHORIZATION='JWT {}'.format(token))
+
+    def _create_token(self):
+        api_key = self.create_api_key(self.user)
+        return self.create_auth_token(api_key.user, api_key.key,
+                                      api_key.secret)
+
+    def test_get_user(self):
+        user, _ = self.auth.authenticate(self.request(self._create_token()))
+        assert user == self.user
+
+    def test_unknown_issuer(self):
+        api_key = self.create_api_key(self.user)
+        payload = self.auth_token_payload(self.user, api_key.key)
+        payload['iss'] = 'non-existant-issuer'
+        token = self.encode_token_payload(payload, api_key.secret)
+
+        with self.assertRaises(AuthenticationFailed):
+            self.auth.authenticate(self.request(token))
+
+    def test_deleted_user(self):
+        self.user.update(deleted=True)
+        with self.assertRaises(AuthenticationFailed):
+            self.auth.authenticate(self.request(self._create_token()))
```

Comment:
A lot of these are missing a check for the exception message itself. Any reason for that? Since `AuthenticationFailed` is raised for so many different cases, I think checking the exception message is useful.